token-lending/program/src/processor.rs

```diff
@@ -1528,6 +1536,142 @@ fn process_set_lending_market_owner(
     Ok(())
 }
 
+#[inline(never)] // avoid stack frame limit
+fn process_flash_loan(program_id: &Pubkey, amount: u64, accounts: &[AccountInfo]) -> ProgramResult {
+    let account_info_iter = &mut accounts.iter();
+    let reserve_liquidity_supply_info = next_account_info(account_info_iter)?;
+    let destination_liquidity_info = next_account_info(account_info_iter)?;
```

Comment:
Should this be checked to be owned by the flash loan fee receiver program?

Reply:
See discussion above.